### PR TITLE
feat(cli): notify when a newer published version is available

### PR DESCRIPTION
**Pull request by chajac**
> [!NOTE]
> PR body AI drafted & edited as needed

# Overview of Changes

The CLI shows its own version with `--version`. It does not tell the user when a newer release exists. Outdated installations stay in use.

This change adds an update notice. When a command starts, the CLI sends a background request to the npm registry for the `latest` dist-tag. When the command completes, the CLI shows a notice if the response is available. The CLI does not wait for the network. If the request fails, the CLI shows nothing.

- The CLI announces each new version one time only. It records the announced version in the `last-update-notice` file in the config directory.
- The notice renders through `ui.note`, so each output mode shows it in its own format. Human mode shows a clack note box. Agent mode writes a two-line plain notice to stderr. JSON mode writes a `{"type":"note",...}` diagnostic line to stderr, the same shape as the other diagnostics. The stdout stream stays unchanged in all modes.
- Set `QAWOLF_NO_UPDATE_CHECK=1` to disable the check. The CLI then sends no request.
- The version comparison is a pure function in `core/version.ts`. The registry request is in `shell/npmRegistry.ts` and does not use the npm executable, so it operates in the standalone binary. The notice policy is in `domains/updateCheck/` with injected dependencies. The file `commands/context.ts` connects the check to the two command wrappers.

Known limits: `--help` and `--version` do not run command actions, so they do not show the notice. A very fast command can complete before the registry response arrives. The CLI then shows no notice for that run. This is the intended trade for zero added latency.

# Testing

```bash
bun run test
bun run typecheck
bun run lint
bun run format:check
bun run knip
```

- New tests cover: version comparison edge cases (prerelease and invalid input give no notice), registry request failure modes, once-per-version behavior across runs, a newer release after a recorded notice, the env opt-out, and an unwritable state file (the notice still shows, no throw).
- Manual checks against the real registry, with a local version below the published version: the box shows in a TTY, the plain notice shows with `--agent`, `--json` writes a note diagnostic to stderr, stdout is unchanged in both, a second run shows nothing, and the notice returns after removal of the state file.

| Mode | stderr | stdout |
|---|---|---|
| Human TTY | `◇ Update available` clack box | unchanged |
| `--agent` | `Update available: @qawolf/cli 1.3.2 → 1.4.0` plus the install line | unchanged |
| `--json` | `{"type":"note","title":"Update available",...}` | unchanged, data only |
| Any mode, second run | nothing (already announced) | unchanged |
| `QAWOLF_NO_UPDATE_CHECK=1` | nothing, and no request sent | unchanged |

# Checklist

- [x] Changes follow the [code style](AGENTS.md) of this project
- [x] Self-review completed
- [x] Tests added/updated
- [x] No breaking changes

### CODE REVIEW

File: src/domains/updateCheck/updateCheck.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        // No notice recorded yet. Fall through and announce.
+      }
+
+      deps.renderNotice(
+        updateCheckMessages.body(deps.currentVersion, latest),
+        updateCheckMessages.title,
+      );
+      try {
+        await deps.fs.mkdir(deps.configDir, { recursive: true });
+        await deps.fs.writeFile(noticePath, `${latest}\n`);
```
[the remark this answers is not in the document]
Comment (chajac) [reply src/domains/updateCheck/updateCheck.ts:72]: Deferring to keep the current iteration. The only effect is that the CLI shows the update notice two times. No other code reads the marker file. Parallel `flows run` cannot cause this, because worker subprocesses do not start an update check.

A correction needs a lock file, because `Fs.writeFile` accepts only `{ mode }` and the CLI updates the marker at each release. Little too much complexity added for a duplicated line.

File: src/domains/updateCheck/updateCheck.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        const notified = (await deps.fs.readFile(noticePath)).trim();
+        if (notified === latest) return;
+      } catch {
+        // No notice recorded yet. Fall through and announce.
+      }
+
+      deps.renderNotice(
+        updateCheckMessages.body(deps.currentVersion, latest),
+        updateCheckMessages.title,
+      );
```
[the remark this answers is not in the document]
Comment (chajac) [reply src/domains/updateCheck/updateCheck.ts]: [7093914](https://github.com/qawolf/cli/pull/1402/commits/70939143a4e42fd7a0a9f35b0a2eb3f819934d86)

File: src/core/version.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * unparseable input return false, so callers stay silent rather than
+ * mis-notify.
+ */
+export function isNewerVersion(current: string, latest: string): boolean {
+  const cur = parseRelease(current);
+  const next = parseRelease(latest);
+  if (cur === undefined || next === undefined) return false;
+  if (next.major !== cur.major) return next.major > cur.major;
+  if (next.minor !== cur.minor) return next.minor > cur.minor;
+  return next.patch > cur.patch;
```
[the remark this answers is not in the document]
Comment (chajac) [reply src/core/version.ts:24]: The inputs are constrained. One version comes from our own `package.json`. The other comes from the npm `latest` dist-tag for this package. npm validates semantic versions at publish time, so `1.01.0` cannot occur. A patch number above 2^53 is also not a real case.

**Review by MateusZitelli — APPROVED** [review]
